### DISCUSSION TITLE
Wix 3.11 to 4.0 Custom Extension class

### DISCUSSION THREAD

**Discussion by kbhanley** [wixtoolset/issues#7513] [Questions]
We are in the process of upgrading from version 3.11 to 4.0 and have an existing custom extension class that houses our custom actions used for multiple different installer projects. We are having trouble locating the WixExtension class that is inherited based on the version 3 example for setting up a Skeleton Wix Extension. What packages do we need to download, and has the namespace or classes changed? And if there is a better way for setting up a custom extension class using version 4, could you describe how to accomplish that?

**Comment by barnson** [comment]
Many things have changed. Source for the "native" WiX extensions is at https://github.com/wixtoolset/wix/tree/develop/src/ext.
